File: Warehouse-b/src/main/java/com/example/warehouse/service/contract/VehicleService.java
package com.example.warehouse.service.contract;

import com.example.warehouse.entity.Vehicle;
import java.util.List;

public interface VehicleService {
    Vehicle registerVehicle(Vehicle vehicle);

    Vehicle getVehicleById(String vehicleId);

    Vehicle getVehicleByNumber(String vehicleNumber);

    List<Vehicle> getAllVehicles();

    List<Vehicle> getActiveVehicles();

    Vehicle updateVehicleLocation(String vehicleId, Double latitude, Double longitude);

    Vehicle updateVehicleStatus(String vehicleId, boolean active);

    void deleteVehicle(String vehicleId);
}

File: web/src/services/VehicleService.ts
import httpClient from '@/lib/httpClient';
import type { VehicleResponse } from '@/types/api';

export const VehicleService = {
    getAllVehicles: (): Promise<VehicleResponse[]> =>
        httpClient.get('/vehicles'),

    getActiveVehicles: async (): Promise<VehicleResponse[]> => {
        const vehicles = await httpClient.get<VehicleResponse[]>('/vehicles');
        return vehicles.filter((v) => v.active);
    },

    getVehicleById: (vehicleId: string): Promise<VehicleResponse> =>
        httpClient.get(`/vehicles/${vehicleId}`),
};
